File: Composition/CS/Composition/Algorithm/Inheritance/SummingAggregator.cs
using System.Collections.Generic;
using System.Linq;

namespace Algorithm.Inheritance
{
    public class SummingAggregator : PointsAggregator
    {
        public SummingAggregator(IEnumerable<Measurement> measurements) : base(measurements)
        {
        }

        protected override IEnumerable<Measurement> FilterMeasurements(IEnumerable<Measurement> measurements)
        {
            return measurements;
        }

        protected override Measurement AggregateMeasurements(IEnumerable<Measurement> measurements)
        {
            return new Measurement {X = measurements.Sum(m => m.X), Y = measurements.Sum(m => m.Y)};
        }
    }
}

File: Composition/Java/Composition/src/main/java/algorithm/inhereitance/SummingAggregator.java
package algorithm.inhereitance;

import static algorithm.MeasurementUtil.sumX;
import static algorithm.MeasurementUtil.sumY;

import java.util.Collection;

import algorithm.Measurement;

public class SummingAggregator extends PointsAggregator {

	public SummingAggregator(Collection<Measurement> measurements) {
		super(measurements);
	}

	@Override
	protected Collection<Measurement> filterMeasurements() {
		return measurements;
	}

	@Override
	protected Measurement aggregateMeasurements() {
		return new Measurement(sumX(measurements), sumY(measurements));
	}
}
